subs/pantry/src/Pantry/Types.hs

```diff
@@ -148,6 +152,21 @@ data Package = Package
   }
   deriving (Show, Eq)
 
+data PackageCabal = PCCabalFile !TreeEntry -- ^ TreeEntry of Cabal file
```

Comment:
This data type feels wrong to me. What's the case where you need the `PCHpack`? I would have expected a data type closer to:

```haskell
data PackageCabal
  = PCCabalFile !TreeEntry
  | PCHpack
      !TreeEntry -- original hpack file, if actually needed
      !TreeEntry -- generated cabal file
      !Version -- version of hpack used
```

Reply:
Changed the data type accordingly.